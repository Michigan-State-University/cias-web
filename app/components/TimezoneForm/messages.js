import { defineMessages } from 'react-intl';

export const scope = 'app.containers.TimezoneForm';

export default defineMessages({
  timeZoneLabel: {
    id: `${scope}.timeZoneLabel`,
    defaultMessage: 'Timezone',
  },
});
