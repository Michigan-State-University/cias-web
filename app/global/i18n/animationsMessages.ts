import { defineMessages } from 'react-intl';

import { NarratorAnimation } from 'models/Narrator';

export const scope = 'app.global.Animations';

export default defineMessages<NarratorAnimation>({
  uncertain: {
    id: `${scope}.uncertain`,
    defaultMessage: 'Uncertain',
  },
  wave: {
    id: `${scope}.wave`,
    defaultMessage: 'Wave',
  },
  greet: {
    id: `${scope}.greet`,
    defaultMessage: 'Greet',
  },
  otohLeftHand: {
    id: `${scope}.otohLeftHand`,
    defaultMessage: 'On the other hand (left)',
  },
  otohRightHand: {
    id: `${scope}.otohRightHand`,
    defaultMessage: 'On the other hand (right)',
  },
  pointRight: {
    id: `${scope}.pointRight`,
    defaultMessage: 'Point right',
  },
  pointLeft: {
    id: `${scope}.pointLeft`,
    defaultMessage: 'Point left',
  },
  pointUp: {
    id: `${scope}.pointUp`,
    defaultMessage: 'Point up',
  },
  pointDown: {
    id: `${scope}.pointDown`,
    defaultMessage: 'Point down',
  },
  pointLeftUp: {
    id: `${scope}.pointLeftUp`,
    defaultMessage: 'Point left up',
  },
  pointRightUp: {
    id: `${scope}.pointRightUp`,
    defaultMessage: 'Point right up',
  },
  explain: {
    id: `${scope}.explain`,
    defaultMessage: 'Explain',
  },
  rest: {
    id: `${scope}.rest`,
    defaultMessage: 'Rest',
  },
  restShoulderRubbing: {
    id: `${scope}.restShoulderRubbing`,
    defaultMessage: 'Rest shoulder rubbing',
  },
  restStretching: {
    id: `${scope}.restStretching`,
    defaultMessage: 'Rest stretching',
  },
  restWeightShift: {
    id: `${scope}.restWeightShift`,
    defaultMessage: 'Rest weight shift',
  },
  read: {
    id: `${scope}.read`,
    defaultMessage: 'Read',
  },
  reading: {
    id: `${scope}.reading`,
    defaultMessage: 'Reading',
  },
  confused: {
    id: `${scope}.confused`,
    defaultMessage: 'Confused',
  },
  doMagic: {
    id: `${scope}.doMagic`,
    defaultMessage: 'Do magic',
  },
  getAttention: {
    id: `${scope}.getAttention`,
    defaultMessage: 'Get attention',
  },
  search: {
    id: `${scope}.search`,
    defaultMessage: 'Search',
  },
  suggest: {
    id: `${scope}.suggest`,
    defaultMessage: 'Suggest',
  },
  surprised: {
    id: `${scope}.surprised`,
    defaultMessage: 'Surprised',
  },
  congratulate: {
    id: `${scope}.congratulate`,
    defaultMessage: 'Congratulate',
  },
  flyIn: {
    id: `${scope}.flyIn`,
    defaultMessage: 'Fly in',
  },
  flyOut: {
    id: `${scope}.flyOut`,
    defaultMessage: 'Fly out',
  },
  standStill: {
    id: `${scope}.standStill`,
    defaultMessage: 'Stand still',
  },
  process: {
    id: `${scope}.process`,
    defaultMessage: 'Process',
  },
  listen: {
    id: `${scope}.listen`,
    defaultMessage: 'Listen',
  },
  think: {
    id: `${scope}.think`,
    defaultMessage: 'Think',
  },
  write: {
    id: `${scope}.write`,
    defaultMessage: 'Write',
  },
  idle: {
    id: `${scope}.idle`,
    defaultMessage: 'Idle',
  },
  turnPage: {
    id: `${scope}.turnPage`,
    defaultMessage: 'Turn page',
  },
  openBook: {
    id: `${scope}.openBook`,
    defaultMessage: 'Open book',
  },
  closeBook: {
    id: `${scope}.closeBook`,
    defaultMessage: 'Close book',
  },
  announce: {
    id: `${scope}.announce`,
    defaultMessage: 'Announce',
  },
  blink: {
    id: `${scope}.blink`,
    defaultMessage: 'Blink',
  },
  acknowledge: {
    id: `${scope}.acknowledge`,
    defaultMessage: 'Acknowledge',
  },
  acknowledgeThoughtful: {
    id: `${scope}.acknowledgeThoughtful`,
    defaultMessage: 'Acknowledge thoughtful',
  },
  browsUp: {
    id: `${scope}.browsUp`,
    defaultMessage: 'Brows up',
  },
  decline: {
    id: `${scope}.decline`,
    defaultMessage: 'Decline',
  },
  declineAnnoyed: {
    id: `${scope}.declineAnnoyed`,
    defaultMessage: 'Decline annoyed',
  },
  declineThoughtful: {
    id: `${scope}.declineThoughtful`,
    defaultMessage: 'Decline thoughtful',
  },
  pleased: {
    id: `${scope}.pleased`,
    defaultMessage: 'Pleased',
  },
  sad: {
    id: `${scope}.sad`,
    defaultMessage: 'Sad',
  },
  hearLeftEar: {
    id: `${scope}.hearLeftEar`,
    defaultMessage: 'Hear left ear',
  },
  hearRightEar: {
    id: `${scope}.hearRightEar`,
    defaultMessage: 'Hear right ear',
  },
  hearBothEars: {
    id: `${scope}.hearBothEars`,
    defaultMessage: 'Hear both ears',
  },
  yawn: {
    id: `${scope}.yawn`,
    defaultMessage: 'Yawn',
  },
  eatCracker: {
    id: `${scope}.eatCracker`,
    defaultMessage: 'Eat cracker',
  },
  wearSunglasses: {
    id: `${scope}.wearSunglasses`,
    defaultMessage: 'Wear sunglasses',
  },
  lookDown: {
    id: `${scope}.lookDown`,
    defaultMessage: 'Look down',
  },
  lookDownAndBlink: {
    id: `${scope}.lookDownAndBlink`,
    defaultMessage: 'Look down and blink',
  },
  glanceUp: {
    id: `${scope}.glanceUp`,
    defaultMessage: 'Glance up',
  },
  glanceDown: {
    id: `${scope}.glanceDown`,
    defaultMessage: 'Glance down',
  },
  glanceLeft: {
    id: `${scope}.glanceLeft`,
    defaultMessage: 'Glance left',
  },
  glanceRight: {
    id: `${scope}.glanceRight`,
    defaultMessage: 'Glance right',
  },
  moveLeft: {
    id: `${scope}.moveLeft`,
    defaultMessage: 'Move left',
  },
  moveRight: {
    id: `${scope}.moveRight`,
    defaultMessage: 'Move right',
  },
});
