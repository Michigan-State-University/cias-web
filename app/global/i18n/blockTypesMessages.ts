import { defineMessages } from 'react-intl';

export const scope = 'app.global.BlockTypes';

export default defineMessages({
  BodyAnimation: {
    id: `${scope}.BodyAnimation`,
    defaultMessage: 'Body Animation',
  },
  Speech: {
    id: `${scope}.Speech`,
    defaultMessage: 'Speech',
  },
  Reflection: {
    id: `${scope}.Reflection`,
    defaultMessage: 'Speech',
  },
  ReflectionFormula: {
    id: `${scope}.ReflectionFormula`,
    defaultMessage: 'Speech',
  },
  HeadAnimation: {
    id: `${scope}.HeadAnimation`,
    defaultMessage: 'Head Animation',
  },
  ReadQuestion: {
    id: `${scope}.ReadQuestion`,
    defaultMessage: 'Read question',
  },
  Pause: {
    id: `${scope}.Pause`,
    defaultMessage: 'Pause',
  },
  Feedback: {
    id: `${scope}.Feedback`,
    defaultMessage: 'Show Spectrum',
  },
});
