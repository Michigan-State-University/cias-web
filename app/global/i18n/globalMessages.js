import { defineMessages } from 'react-intl';

export const scope = 'app.GlobalMessages';

export default defineMessages({
  variables: {
    variableNamePlaceholder: {
      id: `${scope}.variableNamePlaceholder`,
      defaultMessage: 'Variable name...',
    },
    variableScorePlaceholder: {
      id: `${scope}.variableScorePlaceholder`,
      defaultMessage: 'Score',
    },
    value: {
      id: `${scope}.value`,
      defaultMessage: 'Value:',
    },
    emptyBadge: {
      id: `${scope}.emptyBadge`,
      defaultMessage: 'No assigned value',
    },
  },
  blockTypes: {
    BodyAnimation: {
      id: `${scope}.BodyAnimation`,
      defaultMessage: 'Body Animation',
    },
    Speech: {
      id: `${scope}.Speech`,
      defaultMessage: 'Speech',
    },
    Reflection: {
      id: `${scope}.Reflection`,
      defaultMessage: 'Speech',
    },
    HeadAnimation: {
      id: `${scope}.HeadAnimation`,
      defaultMessage: 'Head Animation',
    },
    ReadQuestion: {
      id: `${scope}.ReadQuestion`,
      defaultMessage: 'Read question',
    },
  },
  questionTypes: {
    'Question::Single': {
      id: `${scope}.Question::Single`,
      defaultMessage: 'Single answer screen',
    },
    'Question::Multiple': {
      id: `${scope}.Question::Multiple`,
      defaultMessage: 'Multi answer screen',
    },
    'Question::TextBox': {
      id: `${scope}.Question::TextBox`,
      defaultMessage: 'Textbox screen',
    },
    'Question::Number': {
      id: `${scope}.Question::Number`,
      defaultMessage: 'Number screen',
    },
    'Question::Grid': {
      id: `${scope}.Question::Grid`,
      defaultMessage: 'Grid screen',
    },
    'Question::AnalogueScale': {
      id: `${scope}.Question::AnalogueScale`,
      defaultMessage: 'Visual Analogue Scale screen',
    },
    'Question::Information': {
      id: `${scope}.Question::Information`,
      defaultMessage: 'Information screen',
    },
    'Question::Url': {
      id: `${scope}.Question::Url`,
      defaultMessage: 'Url screen',
    },
    'Question::FollowUpContact': {
      id: `${scope}.Question::FollowUpContact`,
      defaultMessage: 'FollowUpContact screen',
    },
  },
  statuses: {
    draft: {
      id: `${scope}.draft`,
      defaultMessage: 'Draft',
    },
    published: {
      id: `${scope}.published`,
      defaultMessage: 'Published',
    },
    closed: {
      id: `${scope}.closed`,
      defaultMessage: 'Closed',
    },
    archived: {
      id: `${scope}.archived`,
      defaultMessage: 'Archived',
    },
  },
  createProblemError: {
    id: `${scope}.createProblemError`,
    defaultMessage: `Couldn't create an intervention`,
  },
  editProblemError: {
    id: `${scope}.editProblemError`,
    defaultMessage: `Couldn't edit an intervention`,
  },
});
