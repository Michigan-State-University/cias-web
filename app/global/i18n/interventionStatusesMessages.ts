import { defineMessages } from 'react-intl';

export const scope = 'app.global.InterventionStatusesMessages';

export default defineMessages({
  draft: {
    id: `${scope}.draft`,
    defaultMessage: 'Draft',
  },
  published: {
    id: `${scope}.published`,
    defaultMessage: 'Published',
  },
  closed: {
    id: `${scope}.closed`,
    defaultMessage: 'Closed',
  },
  archived: {
    id: `${scope}.archived`,
    defaultMessage: 'Archived',
  },
});
