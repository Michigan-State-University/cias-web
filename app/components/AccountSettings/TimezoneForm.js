import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { Formik } from 'formik';
import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

import FormikSelect from 'components/FormikSelect';
import TIMEZONES from 'utils/timezones';

import messages from './messages';

dayjs.extend(utc);
dayjs.extend(timezone);

const timezoneLabel = timeZone =>
  `${timeZone} (UTC ${dayjs()
    .tz(timeZone)
    .format('Z')})`;

const initialValues = user => {
  const { timeZone } = user;
  return {
    timeZone: {
      value: timeZone,
      label: timezoneLabel(timeZone),
    },
  };
};

const TimezoneForm = ({ formatMessage, user, editUser }) => {
  const onSubmit = ({ timeZone }, { setSubmitting }) => {
    const { value } = timeZone || {};
    if (value) editUser({ timeZone: value });
    setSubmitting(false);
  };

  const timezoneOptions = useMemo(
    () =>
      TIMEZONES.map(elem => ({
        value: elem,
        label: timezoneLabel(elem),
      })),
    [TIMEZONES],
  );

  return (
    <Formik initialValues={initialValues(user)} onSubmit={onSubmit}>
      {() => (
        <FormikSelect
          formikKey="timeZone"
          label={formatMessage(messages.timeZoneLabel)}
          options={timezoneOptions}
        />
      )}
    </Formik>
  );
};

TimezoneForm.propTypes = {
  formatMessage: PropTypes.func,
  user: PropTypes.shape({
    timeZone: PropTypes.string,
  }),
  editUser: PropTypes.func,
};

export default TimezoneForm;
