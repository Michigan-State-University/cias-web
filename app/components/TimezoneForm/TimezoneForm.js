import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { Formik } from 'formik';
import dayjs from 'dayjs';

import { TIMEZONES } from 'utils/timezones';

import FormikSelect from 'components/FormikSelect';

import messages from './messages';

const timezoneLabel = (timeZone) =>
  `${timeZone} (UTC ${dayjs().tz(timeZone).format('Z')})`;

const initialValues = (timeZone) => {
  if (!timeZone) return { timeZone: null };
  return {
    timeZone: {
      value: timeZone,
      label: timezoneLabel(timeZone),
    },
  };
};

export const TimezoneForm = ({
  formatMessage,
  timeZone,
  onChange,
  disabled,
  ...props
}) => {
  const onSubmit = (values, { setSubmitting }) => {
    const { value } = values.timeZone || {};
    if (onChange && value) onChange(value);
    setSubmitting(false);
  };

  const timezoneOptions = useMemo(
    () =>
      TIMEZONES.map((elem) => ({
        value: elem,
        label: timezoneLabel(elem),
      })),
    [TIMEZONES],
  );

  return (
    <Formik
      initialValues={initialValues(timeZone)}
      onSubmit={onSubmit}
      enableReinitialize
    >
      {() => (
        <FormikSelect
          columnStyleProps={props}
          formikKey="timeZone"
          label={formatMessage(messages.timeZoneLabel)}
          options={timezoneOptions}
          disabled={disabled}
        />
      )}
    </Formik>
  );
};

TimezoneForm.propTypes = {
  formatMessage: PropTypes.func,
  timeZone: PropTypes.string,
  onChange: PropTypes.func,
  disabled: PropTypes.bool,
};
