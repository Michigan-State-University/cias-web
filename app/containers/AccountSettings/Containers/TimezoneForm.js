import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';
import { compose } from 'redux';
import { Formik } from 'formik';
import { createStructuredSelector } from 'reselect';
import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

import {
  editUserRequest,
  editUserSaga,
  makeSelectUser,
} from 'global/reducers/auth';
import { useInjectSaga } from 'utils/injectSaga';

import FormikSelect from 'components/FormikSelect';
import TIMEZONES from 'utils/timezones';
import messages from '../messages';

dayjs.extend(utc);
dayjs.extend(timezone);

const timezoneLabel = timeZone =>
  `${timeZone} (UTC ${dayjs()
    .tz(timeZone)
    .format('Z')})`;

const initialValues = user => {
  const { timeZone } = user;
  return {
    timeZone: {
      value: timeZone,
      label: timezoneLabel(timeZone),
    },
  };
};

const TimezoneForm = ({ formatMessage, user, editUser }) => {
  useInjectSaga({ key: 'editUser', saga: editUserSaga });

  const onSubmit = ({ timeZone }, { setSubmitting }) => {
    const { value } = timeZone || {};
    if (value) editUser({ timeZone: value });
    setSubmitting(false);
  };

  const timezoneOptions = useMemo(
    () =>
      TIMEZONES.map(elem => ({
        value: elem,
        label: timezoneLabel(elem),
      })),
    [TIMEZONES],
  );

  return (
    <Formik initialValues={initialValues(user)} onSubmit={onSubmit}>
      {() => (
        <FormikSelect
          formikKey="timeZone"
          label={formatMessage(messages.timeZoneLabel)}
          options={timezoneOptions}
        />
      )}
    </Formik>
  );
};

TimezoneForm.propTypes = {
  formatMessage: PropTypes.func,
  user: PropTypes.shape({
    timeZone: PropTypes.string,
  }),
  editUser: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  user: makeSelectUser(),
});

const mapDispatchToProps = {
  editUser: editUserRequest,
};

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(
  withConnect,
  injectIntl,
)(TimezoneForm);
