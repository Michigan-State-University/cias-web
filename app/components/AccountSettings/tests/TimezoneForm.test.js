/**
 *
 * Tests for AccountSettings
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render, fireEvent, wait } from 'react-testing-library';
import 'jest-styled-components';
import { DEFAULT_LOCALE } from 'i18n';
import { IntlProvider } from 'react-intl';

import TimezoneForm from '../TimezoneForm';

describe('<TimezoneForm />', () => {
  const defaultProps = {
    user: { timeZone: 'America/New_York' },
    editUser: jest.fn(),
    formatMessage: jest.fn(msg => msg.defaultMessage),
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <TimezoneForm {...defaultProps} />
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <TimezoneForm {...defaultProps} />
      </IntlProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  it.skip('Should handle edit correctly', async () => {
    const { getByTestId, debug } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <TimezoneForm {...defaultProps} />
      </IntlProvider>,
    );

    const newTimezone = 'Africa/Harare';
    const timezoneSelect = getByTestId('select');
    fireEvent.click(timezoneSelect);
    debug();
    fireEvent.change(timezoneSelect, {
      value: 'Africa/Casablanca',
      label: 'Africa/Casablanca (UTC +01:00)',
    });
    fireEvent.blur(timezoneSelect);

    await wait(() => {
      expect(defaultProps.editUser).toHaveBeenCalledWith({
        timeZone: newTimezone,
      });
    });
  });
});
