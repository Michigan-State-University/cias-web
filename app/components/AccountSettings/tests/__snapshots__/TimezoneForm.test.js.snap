// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimezoneForm /> Should render and match the snapshot 1`] = `
.c1 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  margin-bottom: 5px;
  color: #2F3850;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c2 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
}

.c3 {
  width: auto;
  height: auto;
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

<div>
  <div
    class="c0"
  >
    <p
      class="c1"
      width="fit-content"
    >
      Time zone
    </p>
    <div
      class="c2"
      data-testid="select"
    >
      <div
        class=" css-2b097c-container"
      >
        <div
          class=" css-4iyz22-control"
        >
          <div
            class=" css-g1d714-ValueContainer"
          >
            <div
              class=" css-1uccc91-singleValue"
            >
              America/New_York (UTC -05:00)
            </div>
            <div
              class="css-b8ldur-Input"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-3-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <img
                class="c3"
                src="IMAGE_MOCK"
                transform=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
